### DISCUSSION TITLE
AWS Secrets Manager As A Config Source

### DISCUSSION THREAD

**Discussion by akil-rails** [quarkusio/quarkus#33026] [Q&A]
Is there a way/plan to support AWS Secrets Manager as a Config Source in addition to Consul, Spring Cloud Config and Vault extensions that are documented [here](https://quarkus.io/guides/config-reference)? Probably something along the lines of Spring's [support](https://cloud.spring.io/spring-cloud-static/spring-cloud-aws/2.1.0.RELEASE/single/spring-cloud-aws.html).

There is a [quarkus extension](https://quarkus.io/extensions/io.quarkiverse.amazonservices/quarkus-amazon-secretsmanager) for connecting to AWS Secrets Manager; Should this be used to create a [Custom ConfigSource?](https://quarkus.io/guides/config-extending-support#custom-config-source)?

**Comment by radcortez** [comment]
> There is a [quarkus extension](https://quarkus.io/extensions/io.quarkiverse.amazonservices/quarkus-amazon-secretsmanager) for connecting to AWS Secrets Manager; Should this be used to create a [Custom ConfigSource?](https://quarkus.io/guides/config-extending-support#custom-config-source)?

It would probably require some changes, because you need configuration to configure the source :)

Are you ok with contributing a source for AWS?

**Reply by akil-rails** [reply]
Yes. Though will need some initial guidance on how to go about it

**Reply by radcortez** [reply]
Thanks! Please, look into https://github.com/quarkiverse/quarkus-config-extensions. It has config extensions to other sources that should help you get started.

**Comment by SebaLopez94** [comment]
Any news about it?

**Reply by radcortez** [reply]
Please check here: https://github.com/quarkiverse/quarkus-amazon-services/pull/771

**Comment by nitinty** [comment]
@akil-rails 
Did you found any workaround for this ?

**Reply by radcortez** [reply]
There was an open PR to add the support for it, but it seems it didn't make it through. Can you help with that work? Thanks!

**Reply by nitinty** [reply]
Can you please share that MR if you have it handy ?

**Reply by radcortez** [reply]
https://github.com/quarkiverse/quarkus-amazon-services/pull/771
